### PR TITLE
Update Client and Server to use StructOpt

### PR DESCRIPTION
**Pull request by mahimna-fb**

### CODE REVIEW

**Comment by mahimna-fb** [comment]
@matbd

File: rust/fastpay/src/client.rs [hunk 1]
```diff
@@ -3,7 +3,6 @@
 
 #![deny(warnings)]
 
-extern crate clap;
 extern crate env_logger;
```
Comment (matbd) [inline rust/fastpay/src/client.rs:6]: FYI we can remove this in the newest Rust versions. (I should will sometime)

File: rust/fastpay/src/client.rs [hunk 2]
```diff
@@ -278,101 +277,113 @@ fn deserialize_response(response: &[u8]) -> Option<AccountInfoResponse> {
...
+
+    let send_timeout = Duration::from_micros(matches.send_timeout);
+    let recv_timeout = Duration::from_micros(matches.recv_timeout);
+    let accounts_config_path = &matches.accounts;
+    let committee_config_path = &matches.committee;
+    let buffer_size = matches.buffer_size.parse::<usize>().unwrap();
 
     let mut accounts_config =
-        AccountsConfig::read_or_create(accounts_config_path).expect("Unable to read user accounts");
+        AccountsConfig::read_or_create(&accounts_config_path).expect("Unable to read user accounts");
```
Comment (matbd) [inline rust/fastpay/src/client.rs]: This `&` is surprising because `&matches.accounts` is already a reference.

File: rust/fastpay/src/server.rs [hunk 3]
```diff
@@ -218,27 +228,24 @@ fn main() {
...
+        ServerCommands::Generate { protocol, host, port, shards } => {
+            let network_protocol = protocol.parse().unwrap();
             let (address, key) = get_key_pair();
             let authority = AuthorityConfig {
                 network_protocol,
                 address,
-                host: subm.value_of("host").unwrap().parse().unwrap(),
-                base_port: subm.value_of("port").unwrap().parse().unwrap(),
-                num_shards: subm.value_of("shards").unwrap().parse().unwrap(),
+                host: host,
```
Comment (matbd) [inline rust/fastpay/src/server.rs:251]: Here you can write `host,` instead of `host: host,`

**Review by matbd — APPROVED** [review]
nice!

File: rust/fastpay/src/client.rs (hunk 2 above)
Comment (mahimna-fb) [reply rust/fastpay/src/client.rs]: good point. it works without the & also so i'll remove it

**Pull request merged**